_posts/2025-02-01-next-scala-lts.md

```diff
@@ -0,0 +1,117 @@
+by: Tomasz Godzik
+---
+
+**TLDR;**
+
+API used in lazy val implementation, will be removed in a future JDK thus one of
+the Scala 3 Next minors in 2025 Q4 and the next LTS will drop JDK 8 support.
+
+## Intro
+
+For the compiler’s second LTS version, the Scala 3 compiler team will be
+dropping support for JDK 8 and is considering which later JDK to use.
+
+## Immediate motivation
+
+One of the main reasons is that recent [JEP 471](https://openjdk.org/jeps/471)
+stated that the memory-access methods in sun.misc.Unsafe are scheduled for
+removal in a future release. Currently, Scala 3 uses sun.misc.Unsafe in its
+implementation of lazy values. This was needed due to compatibility with JDK 8.
+In order to support later versions of JDK we will need to drop usage of Unsafe,
+which is being investigated under
+[this issue](https://github.com/scala/scala3/issues/9013). It also started to
+dawn on the compiler team that we might consider stopping to support JDK 8
+altogether in a future release of Scala 3.
+
+## Long term motivation
+
+Switching to a newer version of the JDK would allow both the compiler and the
+Scala ecosystem to start using new features and standard library improvements
+brought in by JDK 9+. This might include a number of JEPs, which will be
+investigated by the team the coming year. To check the current status take a
+look at the issues marked with the
+[JEP label](https://github.com/scala/scala3/issues?q=is%3Aissue+is%3Aopen+label%3Acompat%3Ajava%3Ajep).
+
+Some of the features that might be interesting to the Scala community are:
+
+- [JEP 181](https://github.com/scala/scala3/issues/22292) will allow classes
+  that are logically part of the same code entity, but which are compiled to
+  distinct class files, to access each other's private members without the need
+  for Scala compiler to insert accessibility-broadening bridge methods. This is
+  especially useful for nested classes or companions.
+- [JEP 409](https://github.com/scala/scala3/issues/22298) which introduces
+  sealed classes and would allow the compiler to better represent the Scala's
+  sealed trait hierarchy.
+- [JEP 441](https://github.com/scala/scala3/issues/22450) would similarily allow
+  us to optimize code resulting from pattern matches using the introduced
+  pattern matching for switch statements in Java.
+- [JEP 471](https://github.com/scala/scala3/issues/9013) which adds a new API
+  for memory access, which can be used to replace the usage of sun.misc.Unsafe
+  in Scala 3.
+
+Another advantage would be reducing the maintenance burden on tooling and
+library authors that currently have to take into account a large number of
+different versions to test and make sure that their code is performant on all of
+them. Thus, the benefits of this change will be felt ecosystem-wide.
+
+If we take a look at the timeline we'll see that JDK 8 was first published on
+18th March 2014, which is already over 10 years ago. Technology and especially
```

Comment:
Better to link to the eol https://www.oracle.com/java/technologies/java-se-support-roadmap.html

Reply:
Added